### DISCUSSION TITLE
Making TrinityCore work with XAMPP and MariaDB

### DISCUSSION THREAD

**Discussion by fosley** [TrinityCore/TrinityCore#27826] [Guides]
TrinityCore doesn't natively work well with MariaDB, but I recently got the Trinity-Bots branch (main project found <a href='https://github.com/trickerer/Trinity-Bots'>here</a>, merged project found <a href='https://github.com/trickerer/TrinityCore-3.3.5-with-NPCBots'>here</a>) working with the MariaDB version I'm using with XAMPP, so I thought I'd post my experiences in the hopes it might help someone else.

The full walkthrough with pictures is on the Trinity-Bots page, <a href='https://github.com/trickerer/Trinity-Bots/discussions/166'>here</a>.

Basic steps:
1. Download the full MariaDB version from <a href='https://mariadb.org/download'>here</a> if you don't have it (XAMPP, for example, ships with a trimmed-down version that doesn't have the necessary dev files). Make sure to download whatever major version you're using on your server. My server is **10.4**.17, so I downloaded **10.4**.24.
2.  If you're using MariaDB 10.4.24 on Windows 64-bit, you can download the _libmariadb_ files I already compiled here: [libmariadb 10.4.24.zip](https://github.com/TrinityCore/TrinityCore/files/8138794/libmariadb.10.4.24.zip), then skip to step 8. Otherwise you'll have to compile the files yourself.
3. Go to the MariaDB about page <a href='https://mariadb.com/kb/en/about-mariadb-connector-c/'>here</a> to find out which Connector-C version was shipped with your MariaDB version. I'm using 10.4, so I need Connector-C 3.1.16.
4. Go to the Connector-C Tags page <a href='https://github.com/mariadb-corporation/mariadb-connector-c/tags'>here</a> on GitHub, then find and download the version you need. I would download v3.1.16. Presumably you could also GitClone it straight off the Tags page, but I'm stupid and wasn't sure how to do that.
5. Extract Connector-C and run CMAKE on it to build a solution using your IDE of choice.
6. Edit the file _include\mariadb_version.h_ so all the version number defines match your version. MARIADB_CLIENT_VERSION_STR, MARIADB_VERSION_ID, MYSQL_VERSION_ID, and MYSQL_SERVER_VERSION need to be updated.
7. Compile the Connector-C solution. Look in the _libmariadb\Release_ folder for two files, _libmariadb.lib_ and _libmariadb.dll_.
8. Copy _libmariadb.lib_ to your _MariaDB\lib_ folder, beside _mariadbclient.lib_.
9. GitClone or download the TrinityCore project as usual.
10. Run CMAKE, pointing source to the Core folder and build to your Build folder. Hit Configure once.
11. Scroll down the list until you find the MySQL references. Make sure _MYSQL_EXECUTABLE_ points to XAMPP's _mysql.exe_ binary, _MYSQL_INCLUDE_DIR_ points to MariaDB's _include\mysql_ folder, and _MYSQL_LIBRARY_ points to the _libmariadb.lib_ binary you compiled or downloaded in step 2 or 6.
12. Hit Configure again, then Generate.
13. Open the generated project in your IDE, then compile it as usual.
14. While you're copying server files, like _worldserver.exe_, into your Server folder, copy _libmariadb.dll_ from step 2 or 6 into the Server folder instead of _libmysql.dll_.

Everything else is like the normal TrinityCore setup. I used it with 3.3.5 Trinity-Bots core, but I don't see why it wouldn't work with most other cores.

**Comment by CraftedRO** [comment]
you know that all of those stepps can be skipped only reverting requirements in src\server\database\Database\databaseworkerpool.cpp back to 5.1 then simple compile and run without problems.

![databaseworkerpool](https://user-images.githubusercontent.com/24683355/191134215-26170e58-bc22-4171-9d7c-b7c13514a5a9.PNG)
